### DISCUSSION TITLE
How to drag and drop windows from the client area?

### DISCUSSION THREAD

**Discussion by sudongg** [microsoft/microsoft-ui-xaml#7356] [Q&A]
In wpf I can use Window.DragMove(), D&D window on client area, is there a similar method in WindowsAppSDK?

**Comment by castorix** [comment]
I'm not sure if I have understood how you want to do Drag&Drop, but maybe you can follow the steps from [Drag and drop](https://docs.microsoft.com/en-us/windows/apps/design/input/drag-and-drop)

(quick test with an Image control in a Grid :
![DragDropTest](https://user-images.githubusercontent.com/22345506/178281281-c919ef0b-7c4d-42fc-a6e4-93f2d786dac1.gif)

**Reply by sudongg** [reply]
hello, this is not what i want, i want to drag the window in the client area.

**Reply by castorix** [reply]
> hello, this is not what i want, i want to drag the window in the client area.

I don't understand what you mean : drag&drop is always done in the Client area (the window part excluding the Caption & Borders)

Or another test with PointerPressed/PointerMoved/PointerReleased, I drag&drop an Image or a Rectangle in the Client area =>

![WinUI3_DragDrop](https://user-images.githubusercontent.com/22345506/178346119-e71bc8cf-100b-487b-93e0-f726af7b5b87.gif)

**Reply by sudongg** [reply]
I want to D&D windows, not UI elements.

**Reply by castorix** [reply]
> I want to D&D windows, not UI elements.

It is the same method, with _PointerPressed_, etc...
You can see for example the test sample I had posted at [WinUI3_Transparent](https://github.com/castorix/WinUI3_Transparent)
(transparency by color value is broken since 1.1.0, but dragging works (_root.PointerPressed_, etc... in MainWindow.xaml.cs ))

**Reply by sudongg** [reply]
You are using AppWindow.Move to move the window, which should be similar to the MoveWindow api, which is different from wpf's Window.DragMove() or dragging the window on the title bar, such as moving to the edge of the window without a docking prompt. Movement does not become Normal state when maximized, etc.
The WinUI3_Transparent project gets stuck when dragging the window quickly.

In the past I used in non-wpf projects
`SendMessage(hWnd, WM_NCLBUTTONDOWN, HTCAPTION, MAKELPARAM(0, 0));`
or
`PostMessage((HWND)IntToPtr(windowID), WM_NCLBUTTONDOWN, HTCAPTION, MAKELPARAM(0, 0));`
But it doesn't work in the winui3 project. After clicking, you need to release the mouse to move the window.

**Reply by castorix** [reply]
It is a quick test showing this usual method similar to the MS sample in WPF with an Adorner (on child controls), 
but it must be improved, like by adding CapturePointer/ReleasePointer that I used with child controls
